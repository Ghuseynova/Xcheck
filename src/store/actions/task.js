import {
  GET_TASK_SESSION_BY_ID,
  POST_TASK_SESSIONS,
  UPDATE_TASK_SESSION,
  GET_TASKSTABLE_SESSIONS,
} from './types/task';

const getTaskSessionById = (id) => {
  return {
    type: GET_TASK_SESSION_BY_ID,
    payload: id,
  };
};

const postTaskSession = (values) => {
  return {
    type: POST_TASK_SESSIONS,
    payload: values,
  };
};

const updateTaskSession = (values) => {
  return {
    type: UPDATE_TASK_SESSION,
    payload: values,
  };
};

const getTasksTable = () => {
  return {
    type: GET_TASKSTABLE_SESSIONS,
  };
};

export { getTaskSessionById, postTaskSession, updateTaskSession, getTasksTable };
