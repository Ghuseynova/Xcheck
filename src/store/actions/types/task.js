const GET_TASKSTABLE_SESSIONS = 'GET_TASKSTABLE_SESSIONS';
const GET_TASKSTABLE_SESSIONS_SUCCESS = 'GET_TASKSTABLE_SESSIONS_SUCCESS';
const GET_TASKSTABLE_SESSIONS_FAILURE = 'GET_TASKSTABLE_SESSIONS_FAILURE';

const GET_TASK = 'GET_TASK';
const GET_TASK_SUCCESS = 'GET_TASK_SUCCESS';
const GET_TASK_FAILURE = 'GET_TASK_FAILURE';

const UPDATE_TASK_SESSION = 'UPDATE_TASK_SESSION';
const UPDATE_TASK_SESSION_SUCCESS = 'UPDATE_TASK_SESSION_SUCCESS';
const UPDATE_TASK_SESSION_FAILURE = 'UPDATE_TASK_SESSION_FAILURE';

const POST_TASK_SESSIONS = 'POST_TASK_SESSIONS';
const POST_TASK_SESSIONS_SUCCESS = 'POST_TASK_SESSIONS_SUCCESS';
const POST_TASK_SESSIONS_FAILURE = 'POST_TASK_SESSIONS_FAILURE';

export {
  GET_TASKSTABLE_SESSIONS,
  GET_TASKSTABLE_SESSIONS_SUCCESS,
  GET_TASKSTABLE_SESSIONS_FAILURE,
  POST_TASK_SESSIONS,
  POST_TASK_SESSIONS_SUCCESS,
  POST_TASK_SESSIONS_FAILURE,
  GET_TASK,
  GET_TASK_SUCCESS,
  GET_TASK_FAILURE,
  UPDATE_TASK_SESSION,
  UPDATE_TASK_SESSION_SUCCESS,
  UPDATE_TASK_SESSION_FAILURE,
};
