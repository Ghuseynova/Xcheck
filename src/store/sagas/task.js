import { takeEvery, call, put } from 'redux-saga/effects';
import { axiosDB } from '../../axios';
import {
  POST_TASK_SESSIONS,
  POST_TASK_SESSIONS_SUCCESS,
  POST_TASK_SESSIONS_FAILURE,
  UPDATE_TASK_SESSION_SUCCESS,
  UPDATE_TASK_SESSION_FAILURE,
  UPDATE_TASK_SESSION,
  GET_TASKSTABLE_SESSIONS,
  GET_TASKSTABLE_SESSIONS_SUCCESS,
  GET_TASKSTABLE_SESSIONS_FAILURE,
  GET_TASK_SESSION_BY_ID,
} from '../actions/types/task';

function* workerPostTask(action) {
  const uri = 'http://localhost:3000/tasks';
  try {
    yield axiosDB.post('tasks', action.payload);
    yield put({ type: POST_TASK_SESSIONS_SUCCESS });
  } catch {
    yield put({
      type: POST_TASK_SESSIONS_FAILURE,
      payload: `ERROR! Cannot post task at ${uri}`,
    });
  }
}

function* workerUpdateTaskSession(action) {
  try {
    yield axiosDB.put(`/tasks/${action.payload.id}`, action.payload);
    yield put({ type: UPDATE_TASK_SESSION_SUCCESS });
  } catch {
    yield put({
      type: UPDATE_TASK_SESSION_FAILURE,
      payload: `ERROR! Cannot update task session with this ID`,
    });
  }
}

function* workerGetTasksTable() {
  const uri = 'http://localhost:3000/tasks';
  try {
    const result = yield axiosDB.get('tasks');
    yield put({ type: GET_TASKSTABLE_SESSIONS_SUCCESS, payload: result.data });
  } catch {
    yield put({
      type: GET_TASKSTABLE_SESSIONS_FAILURE,
      payload: `ERROR! Cannot get tasks at ${uri}`,
    });
  }
}

function* workerGetTaskTableById(action) {
  const uri = `http://localhost:3000/tasks?taskId=${action.payload}`;
  try {
    const result = yield axiosDB.get(`/tasks?taskId=${action.payload}`);
    yield put({ type: GET_TASKSTABLE_SESSIONS_SUCCESS, payload: result.data });
  } catch {
    yield put({
      type: GET_TASKSTABLE_SESSIONS_FAILURE,
      payload: `ERROR! Cannot get task at ${uri}`,
    });
  }
}

function* watchTask() {
  yield takeEvery(POST_TASK_SESSIONS, workerPostTask);
  yield takeEvery(UPDATE_TASK_SESSION, workerUpdateTaskSession);
  yield takeEvery(GET_TASKSTABLE_SESSIONS, workerGetTasksTable);
  yield takeEvery(GET_TASK_SESSION_BY_ID, workerGetTaskTableById);
}

export default watchTask;
