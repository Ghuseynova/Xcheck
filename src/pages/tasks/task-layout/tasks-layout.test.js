import React from 'react';
import { mount } from 'enzyme';
import { Router } from "react-router-dom";
import { createMemoryHistory } from 'history';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import TaskLayout from './task-layout';
import task from '../test-data';

describe('TaskLayout with id=task-form', () => {
	let store;
	const defaultProps = {}
	beforeEach(() => {
		store = configureStore()(defaultProps);
	});

	const wrapper = (props) => mount(
		<Router history={createMemoryHistory()}>
			<Provider store={store}>
				<TaskLayout {...defaultProps} {...props} />
			</Provider>
		</Router>
	);

	it('toMatchSnapshot TaskLayout with id=task-form', () => {
		expect(wrapper({
			match: { params: { id: 'task-form' } }
		})).toMatchSnapshot();
	});
});

describe('TaskLayout with id=task-edit-form', () => {
	let store;
	const defaultProps = {};
	const onSubmitButton = jest.fn();
	beforeEach(() => {
		store = configureStore()({ tasks: { formValues: task.formValue } });
	});

	const wrapper = (props) => mount(
		<Router history={createMemoryHistory()}>
			<Provider store={store}>
				<TaskLayout {...defaultProps} {...props} onSubmit={onSubmitButton} />
			</Provider>
		</Router>
	);

	it('toMatchSnapshot TaskLayout with id=task-edit-form', () => {
		expect(wrapper({
			match: { params: { id: 'task-edit-form' } }, task
		})).toMatchSnapshot();
	});
});

describe('TaskLayout defaul value', () => {
	let store;
	const defaultProps = {}
	beforeEach(() => {
		store = configureStore()({ tasks: { formValues: task.formValue } });
	});

	const wrapper = (props) => mount(
		<Router history={createMemoryHistory()}>
			<Provider store={store}>
				<TaskLayout {...defaultProps} {...props} />
			</Provider>
		</Router>
	);

	it('toMatchSnapshot TaskLayout defaul value', () => {
		expect(wrapper({
			match: { params: { id: '1' } }
		})).toMatchSnapshot();
	});
});