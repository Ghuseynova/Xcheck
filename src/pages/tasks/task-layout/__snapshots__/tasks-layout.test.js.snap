// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskLayout defaul value toMatchSnapshot TaskLayout defaul value 1`] = `ReactWrapper {}`;

exports[`TaskLayout with id=task-edit-form toMatchSnapshot TaskLayout with id=task-edit-form 1`] = `ReactWrapper {}`;

exports[`TaskLayout with id=task-form toMatchSnapshot TaskLayout with id=task-form 1`] = `ReactWrapper {}`;
